### DISCUSSION TITLE
WPF怎么理解窗口的宽高和屏幕分辨率适配和对应的问题？

### DISCUSSION THREAD

**Discussion by notplay3** [BYJRK/DotNet-Discussions#30] [WPF 开发]
怎么做到设置窗口宽高达到窗口刚好覆盖全屏？

**Comment by BYJRK** [comment]
如果你只是想让窗口铺满屏幕，那么可以直接 `this.WindowState = WindowState.Maximized`

**Reply by notplay3** [reply]
这个很管用

**Reply by notplay3** [reply]
![image](https://user-images.githubusercontent.com/57010379/219684845-9df380b2-aa61-4625-bd3f-6c1877ffa9c7.png)
当我这样设置时，窗体会完全覆盖全屏，
我想自己写窗体的标题栏，但这样设置窗体不会露出哦底部的任务栏，怎么才能达到WindowStyle="SingleBorderWindow"
    WindowState="Maximized"的那种窗体显示效果？

**Reply by BYJRK** [reply]
或许你可以考虑用 WindowChrome 覆写窗口样式，然后继续用Maximaized（我没试过，只是感觉或许可以

**Reply by Dynesshely** [reply]
> ![image](https://user-images.githubusercontent.com/57010379/219684845-9df380b2-aa61-4625-bd3f-6c1877ffa9c7.png) 当我这样设置时，窗体会完全覆盖全屏， 我想自己写窗体的标题栏，但这样设置窗体不会露出哦底部的任务栏，怎么才能达到WindowStyle="SingleBorderWindow" WindowState="Maximized"的那种窗体显示效果？

如果你是单纯想要自定义标题栏的话, 可以使用 WindowChrome 属性自定义, 并且在自定义的同时保留系统的最小化, 最大化和关闭按钮
如果不希望使用系统的最小化, 最大化和关闭按钮, 可以设置 WindowStyle="None", 然后自行实现对应按钮
如果你希望保持全屏, 不给用户最小化或是恢复到正常窗口的选项, 可以设置 WindowStyle="None", WindowState="Maximized", 但同时你需要侦听 alt+space 组合键, 避免用户打开窗口菜单, 同时你还需要占用 win+(上下左右) 键或是在监听到 WindowState 发生变化时设定 WindowState="Maximized"

**Comment by chaojicpu** [comment]
double x = SystemParameters.WorkArea.Width;//得到屏幕工作区域宽度
            double y = SystemParameters.WorkArea.Height;//得到屏幕工作区域高度
            double x1 = SystemParameters.PrimaryScreenWidth;//得到屏幕整体宽度
            double y1 = SystemParameters.PrimaryScreenHeight;//得到屏幕整体高度
            this.Top = 0;
            this.Left = 0;
            this.Width = x1;//设置窗体宽度
            this.Height = y1+1;//设置窗体高度

**Reply by notplay3** [reply]
![image](https://user-images.githubusercontent.com/57010379/219683655-72d3e945-ca55-43cc-b309-ef63b8cf791e.png)
![image](https://user-images.githubusercontent.com/57010379/219683711-b3955851-de2c-42c7-9b9a-706bc6dfcdb9.png)
依然不能完全覆盖，我电脑的分辨率为1920*1080，缩放为100%
